contracts/drafts/issuance/IssuanceAdvanced.sol

```diff
@@ -47,8 +51,8 @@ contract IssuanceAdvanced is Ownable, StateMachine, ReentrancyGuard {
         address _issuanceToken,
         address _currencyToken
     ) public Ownable() StateMachine() {
-        issuanceToken = IERC20Mintable(_issuanceToken);
-        currencyToken = IERC20(_currencyToken);
+        issuanceToken = ERC20MintableDetailed(_issuanceToken);
+        currencyToken = ERC20MintableDetailed(_currencyToken);
```

Comment:
2 /
What is the cost of this writing to storage? Would it be better to save the address, create an interface to this contracts and when using them, instantiate them? I would like to see a comparison for that as well.

Reply:
The tests below, also for `IssuanceAdvanced.sol`:
a. mean average gas used in invest(), with interface var storage, without instantiation when used: 144385
b. mean average gas used in invest(), with address var storage, with interface instantiation when used: 144385